Orange/modelling/fitter.py

```diff
@@ -0,0 +1,94 @@
+from collections import namedtuple
+
+from Orange.base import Learner, Model
+
+LearnerTypes = namedtuple('LearnerTypes', ['classification', 'regression'])
+
+
+class Fitter(Learner):
+    """Handle multiple types of target variable with one learner.
+
+    Subclasses of this class serve as a sort of dispatcher. When subclassing,
+    we provide a `LearnerTypes` instance which contain actual learner classes
+    that handle appropriate data types. The fitter can then be used on any
+    data and will delegate the work to the appropriate learner.
+
+    If the learners that handle each data type require different parameters,
+    you should pass in all the possible parameters to the fitter. The fitter
+    will then determine which parameters have to passed to individual learners.
+
+    """
+    __fits__ = None
+    __returns__ = Model
+
+    # Constants to indicate what kind of problem we're dealing with
+    CLASSIFICATION, REGRESSION = range(2)
+
+    def __init__(self, *args, preprocessors=None, **kwargs):
+        super().__init__(preprocessors=preprocessors)
+        self.args = args
+        self.kwargs = kwargs
+        # Make sure to pass preprocessor params to individual learners
+        self.kwargs['preprocessors'] = preprocessors
+        self.problem_type = self.CLASSIFICATION
```

Comment:
Set this to `None` initially if it's bound to be reset on some condition anyway.